ext/updater.go

```diff
@@ -352,27 +353,41 @@ func (u *Updater) StartServer(opts WebhookOpts) error {
 	}
 
 	var tls bool
-	if opts.CertFile == "" && opts.KeyFile == "" {
+	switch {
+	case opts.CertFile == "" && opts.KeyFile == "":
 		tls = false
-	} else if opts.CertFile != "" && opts.KeyFile != "" {
+	case opts.CertFile != "" && opts.KeyFile != "":
 		tls = true
-	} else {
+	default:
 		return ErrMissingCertOrKeyFile
 	}
 
+	var network string
+	if opts.UnixListen != "" {
+		network = "unix"
+	} else {
+		network = "tcp"
+	}
+
 	u.webhookServer = &http.Server{
 		Addr:              opts.GetListenAddr(),
 		Handler:           u.serveMux,
 		ReadTimeout:       opts.ReadTimeout,
 		ReadHeaderTimeout: opts.ReadHeaderTimeout,
 	}
 
+	var err error
+	var ln net.Listener
+	if ln, err = net.Listen(network, u.webhookServer.Addr); err != nil {
+		panic("listening on " + network + ":" + u.webhookServer.Addr + " failed: " + err.Error())
+	}
```

Comment:
This inlined if statement doesn't really sense given all the values are used outside of it, and it should return an error rather than a panic.
Also, we should make sure to close the network with a defer.

```suggestion
	ln, err := net.Listen(network, u.webhookServer.Addr);
	if  err != nil {
		return fmt.Errorf("failed to listen on %s:%s: %w", network, u.webhookServer.Addr, err)
	}
	
	defer ln.Close()
```

Reply:
> This inlined if statement doesn't really sense

I agree

> Also, we should make sure to close the network with a defer.

`defer ln.Close()` is wrong here, as in the next step we are spawning a goroutine that uses `ln` and then immediately `return nil`, thus executing the deferred `ln.Close()`. We need `ln` to stay open for `Serve()` to work.

Also, we should not add a `defer ln.Close()` in the anonymous function, because `http.Serve()` does it and already go to great lengths to avoid closing the listener twice (see [code](https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/net/http/server.go;l=3488;drc=f721fa3be9bb52524f97b409606f9423437535e8;bpv=0;bpt=0) and issue golang/go#24803).

Reply:
> it should return an error rather than a panic.

Agree. I was influenced by the later `panic("http server failed: " + err.Error())`, but here the situation is different, we still have the chance to return an error.